### DISCUSSION TITLE
New Event Triggers

### DISCUSSION THREAD

**Discussion by TLMondo** [directus/directus#20295] [Feature Requests]
### Summary

There are a few Event Triggers that are missing and would help with added health monitoring.

- `auth.failure`: Allows us to trigger a workflow for failed logins. Ideally we'd update a specific collection with a counter, using the IP and maybe UA as a UUID, and check if the same machine has failed multiple times, and perform our own internal alerts.
- `collection.create`, `collection.update`, `collection.delete`: Allows us to create a blocker for standards enforcement, or trigger internal notifications for review.
- `settings.<crud>`: In instances where there are multiple admins, this allows us to be more aware of changes outside of assumed internal communications. Also can be blocking to allow standards enforcements.
- `webhooks.<crud>`: Any outbound data is VERY important to be aware of, and enforce standards.
- `access_controls.<crud>`: I'd want to be able to create blocking flows that restrict allowing certain roles access to sensitive data, unless done by a "super admin". Also include: `new role`, `invited users`
- `extensions.activated`, `extensions.disabled`: So that we know when an extension comes into play. Would be great to have contextual information as part of the payload that calls out capabilities the extension uses (e.g., File system access, DB access, Users, etc.)

### Basic Example

See above

### Motivation

See above

### Detailed Design

See above

### Requirements List

See above

### Drawbacks

Additional development time?

### Alternatives

A working example of a `Cron Trigger` that performs a lookup for changes in the above examples from the `directus_activity` collection, and then triggers a generic "system event" webhook back into the platform, that we can then process. This would prohibit a _blocking_ flow though.

### Adoption Strategy

No effect to roll out.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
